### ISSUE TITLE
Message payload truncated

### ISSUE THREAD

**Issue by andypiper** [emqtt/CocoaMQTT#7]
When receiving messages (from e.g. mosquitto_pub), the first two characters of the message body are being lost. Messages sent from another CocoaMQTT client are received intact.

**Comment by erylee** [comment]
@andypiper, please check 4513742a8ba7de8360178f6fb3531a66854c3d4f commit, @turtleDeng fixed this issue?

**Comment by erylee** [comment]
CocoaMQTTFramePublish pack/unpack error, Fixed in 1.0 release:

```
class CocoaMQTTFramePublish: CocoaMQTTFrame {

    func unpack() {
        //topic
        var msb = data![0], lsb = data![1]
        let len = UInt16(msb) << 8 + UInt16(lsb)
        var pos: Int = 2 + Int(len)
        topic = NSString(bytes: [UInt8](data![2...(pos-1)]), length: Int(len), encoding: NSUTF8StringEncoding) as? String
        
        //msgid
        if qos == 0 {
            msgid = 0
        } else {
            msb = data![pos]; lsb = data![pos+1]
            msgid = UInt16(msb) << 8 + UInt16(lsb)
            pos += 2
        }
        //payload
        let end = data!.count - 1

        payload = [UInt8](data![pos...end])
    }

    override func pack() {
        variableHeader += topic!.bytesWithLength
        if qos > 0 {
            variableHeader += msgid!.hlBytes
        }
    }

}

```
